RecoMuon/MuonIdentification/src/MuonShowerDigiFiller.cc

```diff
@@ -0,0 +1,153 @@
+//
+// Package:    MuonShowerDigiFiller
+// Class:      MuonShowerDigiFiller
+// 
+/**\class MuonShowerDigiFiller MuonShowerDigiFiller.cc RecoMuon/MuonIdentification/src/MuonShowerDigiFiller.cc
+
+ Description: Class filling shower information using DT and CSC digis 
+
+ Implementation:
+     <Notes on implementation>
+*/
+//
+// Original Author:  Carlo Battilana, INFN BO 
+//         Created:  Sat Mar 23 14:36:22 CET 2019
+//
+//
+
+
+// system include files
+
+// user include files
+#include "RecoMuon/MuonIdentification/interface/MuonShowerDigiFiller.h"
+
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+
+//
+// constructors and destructor
+//
+
+MuonShowerDigiFiller::MuonShowerDigiFiller(const edm::ParameterSet& iConfig, edm::ConsumesCollector&& iC) : 
+  m_digiMaxDistanceX(iConfig.getParameter<double>("digiMaxDistanceX"))
+
+{
+
+  edm::InputTag label = iConfig.getParameter<edm::InputTag>("dtDigiCollectionLabel");
+  m_dtDigisToken=iC.consumes<DTDigiCollection>(label);
+
+  label = iConfig.getParameter<edm::InputTag>("cscDigiCollectionLabel");
+  m_cscDigisToken=iC.consumes<CSCStripDigiCollection>(label);
+   
+}
+
+
+MuonShowerDigiFiller::~MuonShowerDigiFiller()
+{
+
+}
+
+
+//
+// member functions
+//
+
+void 
+MuonShowerDigiFiller::getES( const edm::EventSetup& iSetup )
+{
+
+  iSetup.get<MuonGeometryRecord>().get(m_dtGeometry);
+  iSetup.get<MuonGeometryRecord>().get(m_cscGeometry);
+
+}
+
+void 
+MuonShowerDigiFiller::getDigis( edm::Event& iEvent )
+{
+  
+  iEvent.getByToken(m_dtDigisToken,  m_dtDigis);
+  iEvent.getByToken(m_cscDigisToken, m_cscDigis);
+
+}
+
+void 
+MuonShowerDigiFiller::fill( reco::MuonChamberMatch & muChMatch ) const 
+{
+
+  int nDigisInRange = 0;
+
+  // DT chamber
+  if( muChMatch.detector() == MuonSubdetId::DT ) 
+    {
+      double xTrack = muChMatch.x;
+      
+      for (int sl = 1; sl < 4; sl += 2) 
+	{
+	  for (int layer = 1; layer < 5; ++layer) 
```

Comment:
please use ```DTChamberId::maxLayerId ``` and ```maxSuperLayerId``` instead of the magic numbers

Reply:
OK